### DISCUSSION TITLE
网站开发者如何指定 inline 元素

### DISCUSSION THREAD

**Discussion by 117503445** [immersive-translate/immersive-translate#566] [Help] [answered]
### 概述 | Summary

我是网站的开发者，希望将网站适配 沉浸式翻译。网页中有部分图片需要 inline，我为他们添加了类名 `.inline`

用户可以在 沉浸式翻译 的拓展中，添加规则，定义 `.inline` 为 inline 元素

```json
[
  {
    "matches": "https://my-website",
    "extraInlineSelectors": [
      ".inline"
    ]
  }
]
```

我的问题在于如何修改网页，让用户不需要修改配置，拓展也能成功识别到这些应当 inline 的图片。（我不清楚目前默认的 inline 判断规则是怎么样的）

**Comment by theowenyoung** [comment]
大多数情况下我们是根据 元素的样式来确定 是否inline的，所以理论上你用样式 display: inline , 我们是可以识别的。你现在是遇到什么问题了吗？

**Comment by 117503445** [comment]
你可以在 <https://flow-demo.ali.117503445.top/> 查看这个网页

目前的翻译会将 1 个自然段分割为多个部分

![Screenshot_20230609_111554](https://github.com/immersive-translate/immersive-translate/assets/21152034/61835be4-0678-4bb4-941c-f3f3ee62b642)

在我添加以下配置后，自然段就不会被分开了，这是我预期的行为

```json
{
  "extraInlineSelectors": [
    ".inline-img"
  ],
  "matches": "https://flow-pdf.zj.117503445.top:20000/*"
}
```

![Screenshot_20230609_111649](https://github.com/immersive-translate/immersive-translate/assets/21152034/bba3ca61-35a9-41b1-8766-d55cc6f4dd21)

现在我希望用户不修改配置，拓展也能正确识别 inline image。但是我在设置 `.inline-image { display: inline; } ` 后，问题似乎还是没有被解决。

**Reply by theowenyoung** [reply]
确实。目前拓展针对图片做了单独的处理，认为 (width > 48 || height > 48) 的图片会认为是block元素...

可能这里需要改成 &&..?

**Reply by theowenyoung** [reply, marked as answer]
试了一下，就正常了： 

<img width="849" alt="image" src="https://github.com/immersive-translate/immersive-translate/assets/62473795/b6a23dc8-9f28-4fe4-a324-7ecb24588d44">


会包含在下个版本内

**Reply by 117503445** [reply]
谢谢，辛苦了 ：）
